### DISCUSSION TITLE
Need guidance on training xtts_v2 model from scratch in Hindi

### DISCUSSION THREAD

**Discussion by lalit-mohanani** [coqui-ai/TTS#3575] [General Q&A] [answered]
I've been attempting to train the xtts_v2 model on my custom dataset in Hindi without using the xtts checkpoint, but I haven't been successful in getting any meaningful results. It seems I might be missing some crucial steps or configurations. Could someone please provide guidance on how to train the xtts_v2 model from scratch in Hindi?
I would greatly appreciate any assistance or pointers on the necessary preprocessing steps, model architecture adjustments, hyperparameter tuning, or any other relevant details that could help me achieve success in training the xtts_v2 model for Hindi.

**Comment by avii-km** [comment]
were you able to train model for hindi ?

**Reply by lalit-mohanani** [reply, marked as answer]
yes, but not from scrach but finetuned it.

**Reply by avii-km** [reply]
kindly share implementation for fine tuning for hindi lang dataset

**Reply by lalit-mohanani** [reply]
You can refer https://docs.coqui.ai/en/latest/models/xtts.html

**Comment by VivekGangurde94** [comment]
File /opt/conda/lib/python3.10/site-packages/IPython/core/ultratb.py:1131, in VerboseTB.get_records(self, etb, number_of_lines_of_context, tb_offset)
   1129 while cf is not None:
   1130     try:
-> 1131         mod = inspect.getmodule(cf.tb_frame)
   1132         if mod is not None:
   1133             mod_name = mod.__name__

AttributeError: 'tuple' object has no attribute 'tb_frame'
did u got this error while fine tunning

**Reply by lalit-mohanani** [reply]
Nope

**Reply by Aml-Hassan-Abd-El-hamid** [reply]
> File /opt/conda/lib/python3.10/site-packages/IPython/core/ultratb.py:1131, in VerboseTB.get_records(self, etb, number_of_lines_of_context, tb_offset) 1129 while cf is not None: 1130 try: -> 1131 mod = inspect.getmodule(cf.tb_frame) 1132 if mod is not None: 1133 mod_name = mod.**name**
> 
> AttributeError: 'tuple' object has no attribute 'tb_frame' did u got this error while fine tunning

I had the same issue trying to fine-tune for the Egyptian dialect, did you find any solution? @VivekGangurde94
